src/Microsoft.ML.FastTree/TreeEnsembleFeaturizationEstimator.cs

```diff
@@ -0,0 +1,279 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Linq;
+using Microsoft.ML.Data;
+using Microsoft.ML.Runtime;
+
+namespace Microsoft.ML.Trainers.FastTree
+{
+    /// <summary>
+    /// This class encapsulates the common behavior of all tree-based featurizers such as <see cref="FastTreeBinaryFeaturizationEstimator"/>,
+    /// <see cref="FastForestBinaryFeaturizationEstimator"/>, <see cref="FastTreeRegressionFeaturizationEstimator"/>,
+    /// <see cref="FastForestRegressionFeaturizationEstimator"/>, and <see cref="PretrainedTreeFeaturizationEstimator"/>.
+    /// All tree-based featurizers share the same output schema computed by <see cref="GetOutputSchema(SchemaShape)"/>. All tree-based featurizers
+    /// requires an input feature column name and a suffix for all output columns. The <see cref="ITransformer"/> returned by <see cref="Fit(IDataView)"/>
+    /// produces three columns: (1) the prediction values of all trees, (2) the IDs of leaves the input feature vector falling into, and (3)
+    /// the binary vector which encodes the paths to those destination leaves.
+    /// </summary>
+    public abstract class FeaturizationEstimatorBase : IEstimator<TreeEnsembleFeaturizationTransformer>
+    {
+        /// <summary>
+        /// The common options of tree-based featurizations such as <see cref="FastTreeBinaryFeaturizationEstimator"/>, <see cref="FastForestBinaryFeaturizationEstimator"/>,
+        /// <see cref="FastTreeRegressionFeaturizationEstimator"/>, <see cref="FastForestRegressionFeaturizationEstimator"/>, and <see cref="PretrainedTreeFeaturizationEstimator"/>.
+        /// </summary>
+        public class CommonOptions
+        {
+            /// <summary>
+            /// The name of feature column in the <see cref="IDataView"/> when calling <see cref="Fit(IDataView)"/>.
+            /// The column type must be a vector of <see cref="System.Single"/>.
+            /// </summary>
+            public string InputColumnName;
+
+            /// <summary>
+            /// The estimator has three output columns. Their names would be "Trees" + <see cref="OutputColumnsSuffix"/>,
+            /// "Leaves" + <see cref="OutputColumnsSuffix"/>, and "Paths" + <see cref="OutputColumnsSuffix"/>. If <see cref="OutputColumnsSuffix"/>
+            /// is <see langword="null"/>, the output names would be "Trees", "Leaves", and "Paths".
+            /// </summary>
+            public string OutputColumnsSuffix;
```

Comment:
We went away from magic strings in the TextTransform. Previously with `tokens=+`, we produced a new column named `{OutputColName}_TokenizedText`.

For the estimators API we have users directly enter the column name for the tokens.  We may want to do the same for the Trees/Leaves/Paths of the TreeFeat.

Perhaps:
`OutputColumnTreeName`, `OutputColumnLeavesName`, `OutputColumnPathsName`.

Reply:
And don't add that column if they empty or equal to null.
That way you can actually configure which parts of tree structure do you want.

Reply:
Conversation about TextTransform: 
via @rogancarr in https://github.com/dotnet/machinelearning/issues/2957 PR
>  When using `OutputTokens=true`, `FeaturizeText` creates a new column called `${OutputColumnName}_TransformedText`. This isn't really well documented anywhere, and it's odd behavior. I suggest that we make the tokenized text column name explicit in the API.
> 
> My suggestion would be the following:
> - Change `OutputTokens = [bool]` to `OutputTokensColumn = [string]`, and a `string.NullOrWhitespace(OutputTokensColumn)` signifies that this column will not be created.

Reply:
Now we can do optional output columns and custom output column names. Please see tests for examples (or wait for formal API samples).